airbyte-integrations/connectors/source-klarna/source_klarna/source.py

```diff
@@ -0,0 +1,123 @@
+
+
+# Basic full refresh stream
+class KlarnaStream(HttpStream, ABC):
+    def __init__(self, region: str, playground: bool, authenticator: BasicHttpAuthenticator, **kwargs):
+        self.region = region
+        self.playground = playground
+        self.kwargs = kwargs
+        super().__init__(authenticator=authenticator)
+
+    page_size = 500
+
+    @property
+    def url_base(self) -> str:
+        playground_path = 'playground.' if self.playground else ''
+        if self.region == 'eu':
+            endpoint = f"https://api.{playground_path}klarna.com/"
+        else:
+            endpoint = f"https://api-{self.region}.{playground_path}klarna.com/"
+        return endpoint
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        response_json = response.json()
+        if 'pagination' in response_json:
+            if 'next' in response_json['pagination']:
+                parsed_url = urlparse(response_json['pagination']['next'])
+                query_params = parse_qs(parsed_url.query)
+                # noinspection PyTypeChecker
+                return query_params
+        else:
+            return None
+
+    def request_params(
+            self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        if next_page_token:
+            return dict(next_page_token)
+        else:
+            return {"offset": 0, "size": self.page_size}
+
+
+class Payouts(KlarnaStream):
+    """
+    Payouts read from Klarna Settlements API https://developers.klarna.com/api/?json#settlements-api
+    """
+    primary_key = "payout_date"  # TODO verify
+
+    def path(
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "/settlements/v1/payouts"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        payouts = response.json().get("payouts", [])
+        yield from payouts
+
+
+class Transactions(KlarnaStream):
+    """
+    Transactions read from Klarna Settlements API https://developers.klarna.com/api/?json#settlements-api
+    """
+    primary_key = "capture_id"  # TODO verify
+
+    def path(
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "/settlements/v1/transactions"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        transactions = response.json().get("transactions", [])
+        yield from transactions
```

Comment:
Please create a variable and simplify this to be reusable.
```suggestion
    # move this to KlarnaStream
    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
        """
        :return an iterable containing each record in the response
        """
        records = response.json().get(self.endpoint, [])
        yield from records
```

```python
class Transactions(KlarnaStream):
    endpoint = "transactions"
```